src/utils/search_pypi.ts

```diff
@@ -0,0 +1,70 @@
+interface PyPIPackageResult {
+  name: string;
+  package_manager_url: string;
+}
+
+let cachedData: { projects: { name: string }[] } = { projects: [] };
```

Comment:
please read the nextjs caching page https://nextjs.org/docs/app/building-your-application/caching